.github/workflows/docker-build-and-push.yaml

```diff
@@ -0,0 +1,35 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - add-CDCI-artifact-registry  # Change this to your target branch
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Log in to Google Cloud
+      uses: google-github-actions/auth@v1
+      with:
+        credentials_json: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+
+    - name: Get OAuth2 access token and login to Docker
+      run: |
+        ACCESS_TOKEN=$(gcloud auth print-access-token)
```

Comment:
we just need to make sure that it doesn't print anything out into the command line

Reply:
good call, I checked and looks clear